src/pages/Questionnaire.tsx

```diff
@@ -116,7 +140,49 @@ export function Questionnaire() {
   };
 
   const handleSubmit = () => {
-    console.log("Submit questionnaire");
+    const totalScore = calculateTotalScore();
+    localStorage.setItem("questionnaire", JSON.stringify(responses));
+    localStorage.setItem("totalScore", JSON.stringify(totalScore));
+    localStorage.removeItem("currentQuestion");
+    console.log("Submit questionnaire", responses);
+    console.log("Total Score", totalScore);
+  };
+
+  const currentQuestionType = questions[currentQuestion]?.type;
+  const isNextDisabled =
+    currentQuestionType !== Type.RANGE &&
+    (responses[currentQuestion] === undefined ||
+      responses[currentQuestion] === null);
+
+  const calculateTotalScore = () => {
```

Comment:
we should have this function outside of the Questionnaire component, to avoid re-creating this function on every re-render.